tests/performance/density-kwok.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2024 The KubeVirt Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package performance
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+const (
+	vmBatchStartupLimit = 5 * time.Minute
+	defaultVMCount      = 1000
+)
+
+var (
+	vmCount = getVMCount()
+)
+
+var _ = KWOKDescribe("Control Plane Performance Density Testing using kwok", func() {
+	artifactsDir, _ := os.LookupEnv("ARTIFACTS")
+	var (
+		virtClient kubecli.KubevirtClient
+		startTime  time.Time
+		primed     bool
+	)
+
+	BeforeEach(func() {
+		if !flags.DeployFakeKwokNodesFlag {
+			Skip("Skipping test as KWOK flag is not enabled")
+		}
+		virtClient = kubevirt.Client()
+
+		if !primed {
+			By("Create primer VMI")
+			createFakeVMIBatchWithKWOK(virtClient, 1)
+
+			By("Waiting for primer VMI to be Running")
+			waitRunningVMI(virtClient, 1, 1*time.Minute)
+
+			// Leave a two scrape buffer between tests
+			time.Sleep(2 * PrometheusScrapeInterval)
+
+			primed = true
+		}
```

Comment:
Isn't this wrong?
The vmi are created in the specific `It`, or am I missing something?

Reply:
Initially, we are creating a fake vmi to set a reference for the tests and considering using it for data interpolation for subsequent creation tests.
Since the primer has a different purpose, i.e. for initial setup, it's not directly participating in the specific test cases.

Reply:
Sorry, I still don't get it. What's the purpose of the primer vmi?
I cannot see the difference between this primer vmi and the other? Which setup is needed here?
Thanks

Reply:
Primer VMI is not directly tied up with any testcases, it is a part of initial setup. By initial setup I mean, preparing the environment, ensuring the monitoring is stable before the actual tests begin. 
FYI, the other performance density tests also make use of it https://github.com/kubevirt/kubevirt/blob/cb0dc925db2049a97caa4a88a261c1c494b20380/tests/performance/density.go#L69

Reply:
Thank you for the explanation!